### PR TITLE
Spare IrisAO segment numbers testing

### PR DESCRIPTION
**Pull request by ivalaginja**
Upstream for https://github.com/spacetelescope/hicat-package/pull/340

These are changes that will provide us with correctly controllable hardware, which we can then use as a baseline to do any refactoring in the segmented DM controls.

### CODE REVIEW

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: @kjbrooks do we know the "real world" orientation of the JOST IrisAO? Does JOST have a single corner segment in the upwards direction like the spare HiCAT IrisAO here, or one of the flat sides with segments in a straight line? I am not sure what kind of rotations the camera, Sharpcap... introduce, so we might need to check this in the lab right in front of the camera next time somebody is in. We can probably do this bye applying a global defocus on the IrisAO (either by commanding the mirror or by translating the camera).
In any case, we might want to consider to pull this parameter out so that it is changeable. Also, it's only used for the command display, so this is not crucial for the controls, but it's easier to work with the commands if they're oriented correctly.

Comment (kjbrooks) [reply catkit/hardware/iris_ao/segmented_dm_command.py]: For JOST, I think it's flat side up. I did a lot of testing and -90 is correct for JOST so I recommend that you make this a variable instead of making a hard coded change since that will mess up JOST. I am all for looking more into this, but if we want it to work on both of them right now, we will need to make this a variable.

**Review by kjbrooks — CHANGES_REQUESTED** [review]
In order to make this compatible with JOST and HiCAT, I recommend changing the rotation to a parameter that is passed in by the user.

**Comment by ivalaginja** [comment]
@kjbrooks I made it an input parameter now

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:319]: typo. change to "it's"

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:467]: Since `filename_flat` is only used when `apply_flat_map = True`, I recommend changing the default to `None`

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:482]: add that this is only needed if `apply_flat_map=True`

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:212]: same a below, mention that this is only used if `apply_flat_map`

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:225]: same commend as other places, only used if `apply_flat_map=True`

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:236]: set `filename_flat` to `None`

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:166]: This is not segment names. This is the total number of segments in a pupil given the number of rings. Remove #TODO. Clarify parameters description as you see necessary

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py:13]: Since you don't define `FLAT_COMMAND_FILE` here I recommend you just create a dummy string to make it clear what you mean.

**Review by kjbrooks — APPROVED** [review]
I found a few other things I didn't see the first time. Sorry about that, I am not sure how I missed them the first time. I will approve this PR but I am hoping you will make the changes I requested.

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [reply catkit/hardware/iris_ao/segmented_dm_command.py:319]: Are sure? This is not intended to be "it is", just "the/its name".

**Comment by ivalaginja** [comment]
@kjbrooks I addressed all your comments but opened a new PR for them so that I can merge this one without having to wait for another rounds of tests and re-approval. Let me know if you want to see anything else over there:
https://github.com/spacetelescope/catkit/pull/140

**Pull request merged**
